src/seedu/addressbook/data/person/Address.java

```diff
@@ -6,15 +6,12 @@
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
-public class Address {
+public class Address extends Contact {
 
     public static final String EXAMPLE = "123, some street";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
-    public final String value;
```

Comment:
Consider pushing the constructor to the super class Contact. 
When designing classes for inheritance keep in mind conceptually whats similar in these classes. It helps. Rather than just looking at shared variables and methods.